pkg/tfvars/openstack/ports.go

```diff
@@ -32,9 +32,13 @@ func portTargetToTerraformPort(networkClient *gophercloud.ServiceClient, portTar
 		}
 	}
 
-	terraformFixedIPs := make([]terraformFixedIP, len(portTarget.FixedIPs))
-	for i := range terraformFixedIPs {
-		resolvedSubnetID, resolvedNetworkID, err := resolveSubnetFilter(networkClient, networkID, portTarget.FixedIPs[i].Subnet)
+	terraformFixedIPs := make([]terraformFixedIP, 0, len(portTarget.FixedIPs))
+	for _, fixedIP := range portTarget.FixedIPs {
+		subnetFilter := machinev1alpha1.SubnetFilter{
+			ID:   fixedIP.Subnet.ID,
+			Name: fixedIP.Subnet.Name,
+		}
+		resolvedSubnetID, resolvedNetworkID, err := resolveSubnetFilter(networkClient, networkID, subnetFilter)
```

Comment:
I'd rather change the signature of `resolveSubnetFilter` to match the new subnet filter you defined here in Installer. This helper function is meant to translate to Terraform whatever we have here.